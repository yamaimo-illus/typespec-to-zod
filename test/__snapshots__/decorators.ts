import { z } from "zod";

/*
 ██████╗ ██████╗ ███╗   ███╗██████╗  ██████╗ ███╗   ██╗███████╗███╗   ██╗████████╗███████╗
██╔════╝██╔═══██╗████╗ ████║██╔══██╗██╔═══██╗████╗  ██║██╔════╝████╗  ██║╚══██╔══╝██╔════╝
██║     ██║   ██║██╔████╔██║██████╔╝██║   ██║██╔██╗ ██║█████╗  ██╔██╗ ██║   ██║   ███████╗
██║     ██║   ██║██║╚██╔╝██║██╔═══╝ ██║   ██║██║╚██╗██║██╔══╝  ██║╚██╗██║   ██║   ╚════██║
╚██████╗╚██████╔╝██║ ╚═╝ ██║██║     ╚██████╔╝██║ ╚████║███████╗██║ ╚████║   ██║   ███████║
 ╚═════╝ ╚═════╝ ╚═╝     ╚═╝╚═╝      ╚═════╝ ╚═╝  ╚═══╝╚══════╝╚═╝  ╚═══╝   ╚═╝   ╚══════╝
*/
export const schemaTypeSpecDecorators = z.object({
    items: z.array(z.string()).min(1).max(5),
    name: z.string().min(1).max(5),
    age: z.number().int().gte(1).lte(5),
    count: z.number().int().gt(1).lt(5)
});
