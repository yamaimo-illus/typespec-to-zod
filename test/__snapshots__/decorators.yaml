openapi: 3.0.0
info:
  title: (title)
  version: 0.0.0
tags: []
paths: {}
components:
  schemas:
    TypeSpecDecorators:
      type: object
      required:
        - items
        - name
        - age
        - count
      properties:
        items:
          type: array
          items:
            type: string
          minItems: 1
          maxItems: 5
        name:
          type: string
          minLength: 1
          maxLength: 5
        age:
          type: integer
          format: int16
          minimum: 1
          maximum: 5
        count:
          type: integer
          format: int16
          minimum: 1
          exclusiveMinimum: true
          maximum: 5
          exclusiveMaximum: true
